### ISSUE TITLE
[DETECTION] Missed Compiler Detection

### ISSUE THREAD

**Issue by apkunpacker** [rednaga/APKiD#244]
[classes.dex.zip](https://github.com/rednaga/APKiD/files/5623151/classes.dex.zip)

```sh
>  apkid ~/Downloads/classes2.dex 
 [+] APKiD 2.1.1 :: from RedNaga :: rednaga.io
 [*] ~/Downloads/classes2.dex
  |-> compiler : unknown (please file detection issue!)
```
Dex extracted from 
https://play.google.com/store/apps/details?id=com.cerdillac.instories

```sh
$ apkid "Insta Story Art_1.2.5.apk"
[+] APKiD 2.1.1 :: from RedNaga :: rednaga.io
[*] Insta Story Art_1.2.5.apk
 |-> packer : Tencent's Legu
[*] Insta Story Art_1.2.5.apk!assets/audience_network.dex
 |-> anti_debug : Debug.isDebuggerConnected() check
 |-> anti_vm : possible Build.SERIAL check
 |-> compiler : unknown (please file detection issue!)
[*] Insta Story Art_1.2.5.apk!classes2.dex
 |-> compiler : unknown (please file detection issue!)
[*] Insta Story Art_1.2.5.apk!classes3.dex
 |-> compiler : unknown (please file detection issue!)
[*] Insta Story Art_1.2.5.apk!classes.dex
 |-> anti_disassembly : non-zero link offset, non-zero link size
 |-> compiler : dexlib 2.x
 |-> packer : Mobile Tencent Protect
```

Classes2.dex and Classes3.dex's compiler stay undetected

**Comment by apkunpacker** [comment]
[audience_network.zip](https://github.com/rednaga/APKiD/files/5742048/audience_network.zip)

$ apkid audience_network.dex
[+] APKiD 2.1.1 :: from RedNaga :: rednaga.io
[*] audience_network.dex
 |-> anti_vm : possible Build.SERIAL check
 |-> compiler : unknown (please file detection issue!)
$

**Comment by enovella** [comment]
@apkunpacker Can you please edit the ticket title and be more clear? I don't know what's wrong with it. Can you rephrase please?

**Comment by AbhiTheModder** [comment]
> [@AbhiTheModder](https://github.com/AbhiTheModder) Did you fix this?

No, the marker rule has been fixed, however, it is still detected as `unknown` due to the missing new fix for rules regarding the map list order (https://github.com/rednaga/APKiD/pull/435#issuecomment-2664751082). I will address this issue as soon as I have enough time to work with the Android R8 source.

**Comment by AbhiTheModder** [comment]
Ok done https://github.com/rednaga/APKiD/pull/449 :P
